### PR TITLE
[Task-2] Implement EF

### PR DESCRIPTION
**Pull request by Aleksander-Drumev-WA**
Domain models, DbContext and configurations created.

### CODE REVIEW

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System.Threading.Tasks;
+using WA.Pizza.Infrastructure.Data.Models;
+
+namespace WA.Pizza.Infrastructure.Data.Configurations
+{
+    internal class BasketEntityConfiguration : IEntityTypeConfiguration<Basket>
+    {
+        public void Configure(EntityTypeBuilder<Basket> entity)
+        {
+            // Maybe unnecessary because has PK next line.
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs]: you are right

Comment (var1ap-WA) [reply WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs]: remove this config for Id

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            entity
+                .Property(b => b.Id)
+                .IsRequired();
+
+            entity.HasKey(b => b.Id);
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:12]: we need to write base Entity class

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+            entity.HasOne(b => b.User)
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:14]: you have forgotten about FK

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                .WithOne(b => b.Basket);
+
+            entity.Property(b => b.UserId)
+                .IsRequired(false);
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs]: We agreed that it's not required for basket

File: WA.Pizza.Infrastructure/Data/Configurations/OrderEntityConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+{
+    internal class OrderEntityConfiguration : IEntityTypeConfiguration<Order>
+    {
+        public void Configure(EntityTypeBuilder<Order> entity)
+        {
+            entity.HasKey(o => o.Id);
+
+            entity
+                .Property(o => o.CreatedOn)
+                .HasColumnType("DATETIME2")
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Configurations/OrderEntityConfiguration.cs]: no need

File: WA.Pizza.Infrastructure/Data/Configurations/CatalogItemEntityConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+{
+    internal class CatalogItemEntityConfiguration : IEntityTypeConfiguration<CatalogItem>
+    {
+        public void Configure(EntityTypeBuilder<CatalogItem> entity)
+        {
+            entity.HasKey(ci => ci.Id);
+
+            entity
+                .Property(ci => ci.Name)
+                .HasColumnType("VARCHAR")
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Configurations/CatalogItemEntityConfiguration.cs]: no need

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs (hunk 1 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs]: Done.

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs (hunk 3 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:14]: Now it's there.

File: WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs (hunk 4 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs]: Removed.

File: WA.Pizza.Infrastructure/Data/Configurations/OrderEntityConfiguration.cs (hunk 5 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Configurations/OrderEntityConfiguration.cs]: Removed.

File: WA.Pizza.Infrastructure/Data/Configurations/CatalogItemEntityConfiguration.cs (hunk 6 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Configurations/CatalogItemEntityConfiguration.cs]: Removed.

File: WA.Pizza.Core/ConstantValues.cs [hunk 7]
```diff
@@ -0,0 +1,7 @@
+﻿namespace WA.Pizza.Core
+{
+    public static class ConstantValues
+    {
+        public const int CATALOG_ITEM_NAME_MAX_LENGTH = 150;
```
Comment (var1ap-WA) [inline WA.Pizza.Core/ConstantValues.cs]: remove it

File: WA.Pizza.Core/Models/Order.cs [hunk 8]
```diff
@@ -1,17 +1,16 @@
-﻿using System.Collections.Generic;
+﻿using WA.Pizza.Core.ModelConfig;
 
-namespace WA.Pizza.Infrastructure.Data.Models
+namespace WA.Pizza.Core.Models
 {
-    public class Order
+    public class Order : BaseEntity
```
Comment (var1ap-WA) [inline WA.Pizza.Core/Models/Order.cs:5]: You also need `OrderItem` entity

**Pull request merged**